src/servers/src/postgres/types.rs

```diff
@@ -0,0 +1,677 @@
+                            ScalarValue::Date64(data.map(|d| d as i64))
+                        }
+                        _ => {
+                            return Err(invalid_parameter_error(
+                                "invalid_parameter_type",
+                                Some(&format!(
+                                    "Expected: {}, found: {}",
+                                    server_type, client_type
+                                )),
+                            ))
+                        }
+                    }
+                }
+                &Type::INT8 => {
+                    let data = portal.parameter::<i64>(idx)?;
+                    match server_type {
+                        ConcreteDataType::Int8(_) => ScalarValue::Int8(data.map(|n| n as i8)),
+                        ConcreteDataType::Int16(_) => ScalarValue::Int16(data.map(|n| n as i16)),
+                        ConcreteDataType::Int32(_) => ScalarValue::Int32(data.map(|n| n as i32)),
+                        ConcreteDataType::Int64(_) => ScalarValue::Int64(data),
+                        ConcreteDataType::UInt8(_) => ScalarValue::UInt8(data.map(|n| n as u8)),
+                        ConcreteDataType::UInt16(_) => ScalarValue::UInt16(data.map(|n| n as u16)),
+                        ConcreteDataType::UInt32(_) => ScalarValue::UInt32(data.map(|n| n as u32)),
+                        ConcreteDataType::UInt64(_) => ScalarValue::UInt64(data.map(|n| n as u64)),
+                        ConcreteDataType::Timestamp(unit) => match *unit {
+                            TimestampType::Second(_) => ScalarValue::TimestampSecond(data, None),
+                            TimestampType::Millisecond(_) => {
+                                ScalarValue::TimestampMillisecond(data, None)
+                            }
+                            TimestampType::Microsecond(_) => {
+                                ScalarValue::TimestampMicrosecond(data, None)
+                            }
+                            TimestampType::Nanosecond(_) => {
+                                ScalarValue::TimestampNanosecond(data, None)
+                            }
+                        },
+                        ConcreteDataType::DateTime(_) => ScalarValue::Date64(data),
+                        _ => {
+                            return Err(invalid_parameter_error(
+                                "invalid_parameter_type",
+                                Some(&format!(
+                                    "Expected: {}, found: {}",
+                                    server_type, client_type
+                                )),
+                            ))
+                        }
+                    }
+                }
+                &Type::FLOAT4 => {
+                    let data = portal.parameter::<f32>(idx)?;
+                    match server_type {
+                        ConcreteDataType::Float32(_) => ScalarValue::Float32(data),
```

Comment:
Other integral data types such as i8,i32,u32 etc. must be able to cast into float types.